pysr/regressor_sequence.py

```diff
@@ -0,0 +1,320 @@
+                pred = np.concatenate(
+                    [pred, self._regressor.predict(X=pred_data, index=index)], axis=0
+                )
+            return pred
+
+    def _sliding_window(self, X):
+        return np.lib.stride_tricks.sliding_window_view(
+            X.flatten(), self.recursive_history_length * np.prod(X.shape[1])
+        )
+
+    @classmethod
+    def from_file(
+        cls,
+        *args,
+        recursive_history_length: int,
+        **kwargs,
+    ):
+        assert recursive_history_length is not None and recursive_history_length > 0
+
+        model = cls(recursive_history_length=recursive_history_length)
+        model._regressor = PySRRegressor.from_file(*args, **kwargs)
+        return model
+
+    def __repr__(self):
+        return self._regressor.__repr__().replace(
+            "PySRRegressor", "PySRSequenceRegressor", 1
+        )
+
+    def get_best(self, *args, **kwargs):
+        return self._regressor.get_best(*args, **kwargs)
+
+    def refresh(self, *args, **kwargs):
+        return self._regressor.refresh(*args, **kwargs)
+
+    def sympy(self, *args, **kwargs):
+        return self._regressor.sympy(*args, **kwargs)
+
+    def latex(self, *args, **kwargs):
+        return self._regressor.latex(*args, **kwargs)
+
+    def get_hof(self):
+        return self._regressor.get_hof()
+
+    def latex_table(
+        self,
+        *args,
+        **kwargs,
+    ):
+        """Create a LaTeX/booktabs table for all, or some, of the equations.
+
+        Parameters
+        ----------
+        indices : list[int] | list[list[int]]
+            If you wish to select a particular subset of equations from
+            `self.equations_`, give the row numbers here. By default,
+            all equations will be used. If there are multiple output
+            features, then pass a list of lists.
+        precision : int
+            The number of significant figures shown in the LaTeX
+            representations.
+            Default is `3`.
+        columns : list[str]
+            Which columns to include in the table.
+            Default is `["equation", "complexity", "loss", "score"]`.
+
+        Returns
+        -------
+        latex_table_str : str
+            A string that will render a table in LaTeX of the equations.
+        """
+        if self.variable_names is not None:
+            if len(self.variable_names) == 1:
+                variable_names = self.variable_names[0] + "_{t-0}"
+            else:
+                variable_names = [
+                    variable_name + "_{t-0}" for variable_name in self.variable_names
+                ]
+        else:
+            if self.n_features == 1:
+                variable_names = "x_{t-0}"
+            else:
+                variable_names = [f"x{i}_{{t-0}}" for i in range(self.n_features)]
+        return self._regressor.latex_table(
+            *args, **kwargs, output_variable_names=variable_names
+        )
```

Comment:
Why are these `t-0` instead of the actual time?

Reply:
🤦 my bad